src/main/scala/ch/epfl/scala/decoder/BinaryDecoder.scala

```diff
@@ -188,22 +188,35 @@ class BinaryDecoder(using Context, ThrowOrWarn):
     extension (xs: Seq[DecodedVariable])
       def orTryDecode(f: PartialFunction[binary.Variable, Seq[DecodedVariable]]): Seq[DecodedVariable] =
         if xs.nonEmpty then xs else f.applyOrElse(variable, _ => Seq.empty[DecodedVariable])
-    val decodedVariables = variable match
-      // tryDecode {
+    val decodedVariables = tryDecode {
       case Patterns.CapturedVariable(name) =>
         for
-          metSym <- decodedMethod.symbolOpt.toSeq
-          sym <- CaptureCollector.collectCaptures(metSym)
+          metTree <- decodedMethod.treeOpt.toSeq
+          sym <- CaptureCollector.collectCaptures(metTree)
           if name == sym.nameStr
         yield DecodedVariable.CapturedVariable(decodedMethod, sym)
-      // }.orTryDecode {
-      case _ =>
-        for
-          metSym <- decodedMethod.symbolOpt.toSeq
-          localVar <- VariableCollector.collectVariables(metSym)
-          if variable.name == localVar.sym.nameStr && localVar.startLine <= sourceLine && sourceLine <= localVar.endLine
-        yield DecodedVariable.LocalVariable(decodedMethod, localVar.sym)
-      // }
+      case Patterns.This() =>
+        decodedMethod.owner.thisType.toSeq.map(DecodedVariable.This(decodedMethod, _))
+      case Patterns.DollarThis() =>
+        val dollarThis = for
+          classSym <- decodedMethod.owner.companionClassSymbol.toSeq
+          if classSym.isSubClass(defn.AnyValClass)
+          sym <- classSym.declarations.collect {
+            case sym: TermSymbol if sym.isVal && !sym.isMethod => sym
+          }
+        yield DecodedVariable.DollarThis(decodedMethod, sym)
+        dollarThis match
+          case Nil => decodedMethod.owner.thisType.toSeq.map(DecodedVariable.This(decodedMethod, _))
+          case _ => dollarThis
```

Comment:
If I remember correctly this is the `$this` param in a trait static forwarder. it would be clearer to test directly the type of the decoded method:

```scala
decodedMethod match
  case  _: DecodedMethod.TraitStaticForwarder => 
    decodedMethod.owner.thisType.toSeq.map(DecodedVariable.This(decodedMethod, _))
  case _ =>
    // try decode as a ValueClassThis
```